src/nvim/msgpack_rpc/channel.c

```diff
@@ -160,7 +173,7 @@ uint64_t channel_connect(bool tcp, const char *address,
   channel->data.stream.internal_data = channel;
   wstream_init(&channel->data.stream, 0);
   rstream_init(&channel->data.stream, CHANNEL_BUFFER_SIZE);
-  rstream_start(&channel->data.stream, parse_msgpack, channel);
+  rstream_start(&channel->data.stream, receive_msgpack, channel);
```

Comment:
In `channel_from_stdio()` `parse_msgpack` should also be replaced with `receive_msgpack`.